File: 프로그래머스/1/77484. 로또의 최고 순위와 최저 순위/로또의 최고 순위와 최저 순위.cpp
#include <string>
#include <vector>
#include <cmath>
using namespace std;

vector<int> solution(vector<int> lottos, vector<int> win_nums) {
    int hit = 0;
    int blank = 0;
    int arr[46] = {0};
    
    for (int i = 0; i < 6; i++){
        if (lottos[i] == 0)
            blank++;
        arr[lottos[i]] = 1;
    }
    
    for (int i = 0; i < 6; i++){
        if (arr[win_nums[i]] == 1)
            hit++;
    }
    
    int worst = min(6, 7 - hit);
    int best = min(6, 7 - hit - blank);
    
    return {best, worst};
}

File: 프로그래머스/1/77484. 로또의 최고 순위와 최저 순위/로또의 최고 순위와 최저 순위.py
def f(num):
    if 6 - num > 5:
        return 6
    return 6 - num + 1

def solution(lottos, win_nums):
    win_nums = set(win_nums)
    zero = lottos.count(0)
    cnt = 0 
    
    for num in lottos:
         if num in win_nums:
            cnt += 1
            
    return [f(cnt + zero), f(cnt)]